ignite/contrib/metrics/frequency.py

```diff
@@ -0,0 +1,45 @@
+from ignite.metrics import Metric
+from ignite.handlers.timing import Timer
+from ignite.metrics.metric import sync_all_reduce, reinit__is_reduced
+
+
+class FrequencyMetric(Metric):
+    """Provides metrics for the number of examples processed per second.
+
+    Examples:
+
+        .. code-block:: python
+
+            # Compute number of tokens processed
+            wps_metric = FrequencyMetric(output_transformer=lambda x: x['ntokens'])
+            wps_metric.attach(trainer, name='wps')
+            # Logging with TQDM
+            ProgressBar(persist=True).attach(trainer, metric_names=['wps'])
+            # Progress bar will looks like
+            # Epoch [2/10]: [12/24]  50%|█████      , wps=400 [00:17<1:23]
+    """
+
+    def __init__(self, output_transform=lambda x: x, device=None):
+        self._timer = None
+        self._n = None
+        super(FrequencyMetric, self).__init__(output_transform=output_transform, device=device)
+
+    @reinit__is_reduced
+    def reset(self):
+        self.timer = Timer()
+        self._n = 0
+        super(FrequencyMetric, self).reset()
+
+    @reinit__is_reduced
+    def update(self, output):
+        self._n += output
+
+    @sync_all_reduce("_n")
+    def compute(self):
+        return self._n / self.timer.value()
```

Comment:
We also need to think about the behaviour of this metric in distributed setting (as @sync_all_reduce, @reinit__is_reduced are set). 
Following `@sync_all_reduce("_n")`, we will gather across the processes all values of `_n` into a single one with sum op. However, ` self.timer.value()` is the time of a single process. Probably, this is not what we would like to have. Maybe, we need to "gather" it too. This can be done like that:
```python
elapsed = torch.tensor(self.timer.value(), device=self._device)
dist.barrier()    # Need to setup "import torch.distributed as dist"
dist.all_reduce(elapsed)
elapsed = elapsed.item()
return self._n / elapsed
``` 
Something like that.